webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import './ticketStyle.scss';
+
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    isLoaded?: boolean;
+    ticketDetails?: TicketDetails;
+    defaultUserInstanceID?: string;
+    connectedInstances: Instance[];
+    getIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+        error?: any;
+    }>;
+    setTicket?: (ticketDetails: {}) => void;
+}
+
+export type State = {
+    href: string;
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+const isAssigned = ' is assigned';
+const unAssigned = 'Unassigned';
+const jiraTicketTitleMaxLength = 80;
+const statusIndeterminate = 'indeterminate';
+const statusDone = 'done';
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    truncateString(str: string, num: number) {
+        if (num > str.length) {
+            return str;
+        }
+        return `${str.substring(0, num)}...`;
+    }
+
+    constructor(Props: Props) {
+        super(Props);
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+        }
+
+        if (!ticketID && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        this.state = {
+            href: this.props.href,
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    init() {
+        let instanceID = '';
+        if (this.props.connectedInstances.length === 1) {
+            instanceID = this.props.connectedInstances[0].instance_id;
+        } else if (this.props.defaultUserInstanceID) {
+            instanceID = this.props.defaultUserInstanceID;
+        }
+
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
```

Comment:
instead of these, why couldn't we have a function to extract the issue Key and then use in both blocks `getIssueByKey`? I think is easier to read that way.